client/src/components/BiasForm.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+  <form novalidate>
+    <md-field>
+      <label>Bias</label>
+      <md-select v-model="model.bias_model">
+        <md-option
+          v-for="(value, choice) in choices"
+          :key="choice"
+          :value="value">
+          {{choice}}
+        </md-option>
+      </md-select>
+    </md-field>
+    <double-field
+      v-for="(value, param) in model.bias_params"
+      :value="value"
+      :key="param"
+      :param="param"
+      range=false
+      :placeholder="String(defaults[param])"
+      v-model="model.bias_params[param]"/>
+  </form>
+</template>
+
+<script>
+import DoubleField from './DoubleField.vue';
+import BACKEND_CONSTANTS from '../constants/backend_constants';
+
+const biasChoices = {
+  'Tinker (2010)': 'Tinker10',
+  Unbiased: 'UnityBias',
+  'Mo (1996)': 'Mo96',
+  'Jing (1998)': 'Jing98',
+  'Sheth-Tormen (1999)': 'ST99',
+  'Sheth-Mo-Tormen (2001)': 'SMT01',
+  'Seljak (2004) Without Cosmo': 'Seljak04',
+  'Seljack (2004) With Cosmo': 'Seljak04Cosmo',
+  'Mandelbaum (2005)': 'Mandelbaum05',
+  'Pillepich (2010)': 'Pillepich10',
+  'Manera (2010)': 'Manera10',
+  'Tinker (2010) Peak-Background Split': 'Tinker10PBSplit',
+};
+
+export default {
+  name: 'BiasForm',
+  model: {
+    event: 'onChange',
+    prop: 'parent_model',
+  },
+  props: ['parent_model'],
+  data() {
+    return {
+      model: {
+        bias_model: 'Tinker10',
+        bias_params: BACKEND_CONSTANTS.Bias_params.Tinker10,
+      },
+      defaults: { ...BACKEND_CONSTANTS.Bias_params.Tinker10 },
+      choices: biasChoices,
+    };
+  },
+  updated() {
+    this.$emit('onChange', this.model);
+  },
+  watch: {
+    'model.bias_model': function updateOptions(val) {
+      this.model.bias_params = null;
+      this.$nextTick(function () {
+        this.model.bias_params = BACKEND_CONSTANTS.Bias_params[val];
+        this.defaults = BACKEND_CONSTANTS.Bias_params[val];
+      });
```

Comment:
This is throwing an unexpected unnamed function error. See below:

![Error](https://i.imgur.com/W25Nnix.png)

This can be fixed by making it an arrow function though. Just checked it and it seems to be updating fine after modifying it and saving changes across model changes as well. See below for an example:

```js
this.$nextTick(() => {
  this.model.bias_params = BACKEND_CONSTANTS.Bias_params[val];
  this.defaults = BACKEND_CONSTANTS.Bias_params[val];
});
```

Reply:
@aschnapp that would probably be helpful 😁

Reply:
Will resolve this with next push.